src/helpers/utils/removeUndefinedMessages.ts

```diff
@@ -0,0 +1,23 @@
+export const removeUndefinedMessages = (
+  messageHandlers:
+    | {
+        Id: string | undefined;
+        ReceiptHandle: string | undefined;
+      }[]
+    | undefined
+):
+  | {
+      Id: string;
+      ReceiptHandle: string;
+    }[]
+  | undefined =>
+  messageHandlers?.filter(
+    (messageHandler) =>
+      messageHandler.Id !== undefined &&
+      messageHandler.ReceiptHandle !== undefined
+  ) as
+    | {
+        Id: string;
+        ReceiptHandle: string;
+      }[]
+    | undefined;
```

Comment:
```suggestion
    (messageHandler): messageHandler is {
        Id: string;
        ReceiptHandle: string;
      } =>
      messageHandler.Id !== undefined &&
      messageHandler.ReceiptHandle !== undefined
  );
```
to avoid cast.
And you can put
```
{
        Id: string;
        ReceiptHandle: string;
      }
```
in a named type to avoid duplication with the return type of the function